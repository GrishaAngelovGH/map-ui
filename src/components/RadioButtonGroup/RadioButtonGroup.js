import { useState } from 'react'
import ButtonGroup from 'react-bootstrap/ButtonGroup'
import ToggleButton from 'react-bootstrap/ToggleButton'

const RadioButtonGroup = ({ radios }) => {
  const [radioValue, setRadioValue] = useState('1')

  return (
    <ButtonGroup>
      {
        radios.map((radio, idx) => (
          <ToggleButton
            key={idx}
            id={`radio-${idx}`}
            type='radio'
            variant='outline-primary'
            name='radio'
            value={radio.value}
            checked={radioValue === radio.value}
            onChange={(e) => setRadioValue(e.currentTarget.value)}
          >
            {radio.name}
          </ToggleButton>
        ))
      }
    </ButtonGroup>
  )
}

export default RadioButtonGroup