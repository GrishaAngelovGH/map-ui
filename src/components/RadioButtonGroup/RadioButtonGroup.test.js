import { render } from '@testing-library/react'

import RadioButtonGroup from './RadioButtonGroup'

test('should render RadioButtonGroup component', () => {
  const view = render(
    <RadioButtonGroup
      initialValue='1'
      radios={[
        { name: 'First', value: '1' },
        { name: 'Second', value: '2' },
        { name: 'Third', value: '3' },
      ]}
    />
  )

  expect(view).toMatchSnapshot()
})