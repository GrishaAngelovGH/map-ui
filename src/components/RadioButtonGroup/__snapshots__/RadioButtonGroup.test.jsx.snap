// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render RadioButtonGroup component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="btn-group"
        role="group"
      >
        <input
          autocomplete="off"
          checked=""
          class="btn-check"
          id="radio-1"
          name="radio-1-name"
          type="radio"
          value="1"
        />
        <label
          class="btn btn-outline-primary"
          for="radio-1"
          tabindex="0"
        >
          First
        </label>
        <input
          autocomplete="off"
          class="btn-check"
          id="radio-2"
          name="radio-2-name"
          type="radio"
          value="2"
        />
        <label
          class="btn btn-outline-primary"
          for="radio-2"
          tabindex="0"
        >
          Second
        </label>
        <input
          autocomplete="off"
          class="btn-check"
          id="radio-3"
          name="radio-3-name"
          type="radio"
          value="3"
        />
        <label
          class="btn btn-outline-primary"
          for="radio-3"
          tabindex="0"
        >
          Third
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="btn-group"
      role="group"
    >
      <input
        autocomplete="off"
        checked=""
        class="btn-check"
        id="radio-1"
        name="radio-1-name"
        type="radio"
        value="1"
      />
      <label
        class="btn btn-outline-primary"
        for="radio-1"
        tabindex="0"
      >
        First
      </label>
      <input
        autocomplete="off"
        class="btn-check"
        id="radio-2"
        name="radio-2-name"
        type="radio"
        value="2"
      />
      <label
        class="btn btn-outline-primary"
        for="radio-2"
        tabindex="0"
      >
        Second
      </label>
      <input
        autocomplete="off"
        class="btn-check"
        id="radio-3"
        name="radio-3-name"
        type="radio"
        value="3"
      />
      <label
        class="btn btn-outline-primary"
        for="radio-3"
        tabindex="0"
      >
        Third
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
